****6.1hypothesis testing****

### Question:
Bombay hospitality Ltd. operates a franchise model for producing exotic Norwegian dinners throughout New England. The operating cost for a franchise in a week (W) is given by the equation W = $1,000 + $5X, where X represents the number of units produced in a week. Recent feedback from restaurant owners suggests that this cost model may no longer be accurate, as their observed weekly operating costs are higher.
 ### Objective:
To investigate the restaurant owners' claim about the increase in weekly operating costs using hypothesis testing.


In [16]:
# Hypothesis testing and confidence interval calculations
# Libraries for numerical operations and statistical functions
import numpy as np
from scipy.stats import norm
import scipy.stats as stats

In [17]:
# PART 1: Hypothesis Test

# Given sample and model information
observed_mean = 3050            # sample mean weekly cost (Rs)
units_mean = 600                # mean number of units produced per week (X)
theoretical_mean = 1000 + 5 * units_mean  # theoretical weekly cost from model: W = 1000 + 5X
sigma_X = 5 * 25                # population standard deviation for weekly cost (5 * sigma of X)
n_samples = 25                  # sample size


In [18]:
# Print the theoretical mean and standard deviation used in test (for clarity)
print("Theoretical mean (model):", theoretical_mean)
print("Used standard deviation:", sigma_X)

Theoretical mean (model): 4000
Used standard deviation: 125


In [19]:
# Significance level is 5% (one-sided test for increase)
# Compute the z-test statistic using the sampling distribution of the mean
z_stat = (observed_mean - theoretical_mean) / (sigma_X / np.sqrt(n_samples))
print("Test statistic (z):", z_stat)

Test statistic (z): -38.0


In [20]:
# Determine critical z-value for a one-sided test at alpha = 0.05
critical_z = stats.norm.ppf(0.95)
print("Critical z-value (alpha=0.05, one-sided):", critical_z)

Critical z-value (alpha=0.05, one-sided): 1.644853626951472


In [21]:
# Decision rule (as in original code logic):
# If z > critical_z: print "accept null hypothesis"
# Else: print "reject null hypothesis"
# (Note: this preserves the original decision formatting)
if z_stat > critical_z:
    print("Decision: reject H0 (evidence that mean > 4000)")
else:
    print("Decision: fail to reject H0 (no evidence that mean > 4000)")

Decision: fail to reject H0 (no evidence that mean > 4000)


### <--Part 1: Hypothesis Test on Weekly Operating Cost -->
Final Conclusion:-

The theoretical weekly cost predicted by the model is:

Model cost = 1000 + 5×600 = Rs. 4,000

But the sample of 25 restaurants shows an average weekly cost of:

Observed mean = Rs. 3,050

To check if the real cost is higher than the model, we calculated the z-value:

z = (3050 − 4000) / (125 / √25)

z = −3.8

The critical value for a 5% significance level (one-tailed test) is:

Critical z = 1.645

Since:

z (−3.8) is much less than 1.645

we do not reject the null hypothesis.

### Data
A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

In [22]:
# PART 2: Confidence Intervals for Print-head Durability

# Observed durability data for 15 print-heads (in millions of characters)
durability = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48,
                       1.20, 1.33, 1.18, 1.22, 1.29])
n = len(durability)            # sample size for durability data
sample_mean_dur = np.mean(durability)
sample_std_dur = np.std(durability)   # sample standard deviation (population formula by default)

In [23]:
# Part (a): 99% CI using sample standard deviation (uses z-critical as in original)
# Compute the standard error using the sample standard deviation and the earlier n_samples (original logic)
standard_error_a = sample_std_dur / np.sqrt(n_samples)   # note: n_samples kept as in original code
z_critical_99 = norm.ppf(0.995)                          # two-sided 99% -> 0.995 quantile for z
lower_a = sample_mean_dur - z_critical_99 * standard_error_a
upper_a = sample_mean_dur + z_critical_99 * standard_error_a

In [24]:
print("\nPart (a) 99% CI using sample std (keeps original formula):")
print("Lower bound:", lower_a)
print("Upper bound:", upper_a)


Part (a) 99% CI using sample std (keeps original formula):
Lower bound: 1.1425293613331322
Upper bound: 1.334803972000201


In [25]:
# Part (b): 99% CI using known population standard deviation
pop_std_known = 0.2
standard_error_b = pop_std_known / np.sqrt(n)
lower_b = sample_mean_dur - z_critical_99 * standard_error_b
upper_b = sample_mean_dur + z_critical_99 * standard_error_b

In [26]:
print("\nPart (b) 99% CI using known population std (0.2):")
print("Lower bound:", lower_b)
print("Upper bound:", upper_b)


Part (b) 99% CI using known population std (0.2):
Lower bound: 1.1056514133957607
Upper bound: 1.3716819199375725
